### PR TITLE
feat: creation MSSQL Terraform provider

### PR DESCRIPTION
**Pull request by zucchinidev**
### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you ran `make run-integration-tests` and `make run-terraform-tests`?
* [x] Have you ran acceptance tests for the service under change?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: providers/terraform-provider-csbsqlserver/connector/connector_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	port          int
+	adminPassword string
+	db            *sql.DB
+	conn          *connector.Connector
+)
+
+var _ = BeforeSuite(func() {
+	adminPassword = testhelpers.RandomPassword()
+	port = testhelpers.FreePort()
+	session := testhelpers.StartServer(adminPassword, port)
```
Comment (pivotal-marcela-campo) [inline providers/terraform-provider-csbsqlserver/connector/connector_suite_test.go:30]: This StartServer set up creates a database. So strictly speaking we are only testing that we can connect and create bindings when the database is already there. We are missing testing the scenario when the database has not yet been created.

**Review by pivotal-marcela-campo — CHANGES_REQUESTED** [review]

File: providers/terraform-provider-csbsqlserver/connector/connector_suite_test.go (hunk 1 above)
Comment (pivotal-marcela-campo) [reply providers/terraform-provider-csbsqlserver/connector/connector_suite_test.go:30]: Discussed with @zucchinidev and agreed to provide the tests in a separate pull request to unblock other tasks. The scanario missing is tested with acceptance tests as well and those pass, so we have confidence this works.

**Review by pivotal-marcela-campo — APPROVED** [review]

File: providers/terraform-provider-csbsqlserver/go.mod [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+go 1.20
+
+require (
+	github.com/denisenkom/go-mssqldb v0.12.3
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.27.0
+	github.com/maxbrunsfeld/counterfeiter/v6 v6.6.2
+	github.com/onsi/ginkgo/v2 v2.11.0
+	github.com/onsi/gomega v1.27.10
+	github.com/pborman/uuid v1.2.1
```
Comment (fnaranjo-vmw) [inline providers/terraform-provider-csbsqlserver/go.mod:11]: We might want to replace this with `https://github.com/google/uuid`

File: providers/terraform-provider-csbsqlserver/go.mod [hunk 3]
```diff
@@ -0,0 +1,73 @@
+module github.com/cloudfoundry/csb-brokerpak-aws/terraform-provider-csbsqlserver
+
+go 1.20
+
+require (
+	github.com/denisenkom/go-mssqldb v0.12.3
+	github.com/hashicorp/terraform-plugin-sdk/v2 v2.27.0
+	github.com/maxbrunsfeld/counterfeiter/v6 v6.6.2
```
Comment (fnaranjo-vmw) [inline providers/terraform-provider-csbsqlserver/go.mod:8]: I believe we are not using this package.
According to counterfeiter documentation, for it to do anything we need to [add go:generate directives](https://github.com/maxbrunsfeld/counterfeiter/#step-2a---add-gogenerate-directives). 
What makes me think that we don't need it for this project.
I have searched for "go:generate

File: providers/terraform-provider-csbsqlserver/go.mod [hunk 4]
```diff
@@ -0,0 +1,73 @@
+module github.com/cloudfoundry/csb-brokerpak-aws/terraform-provider-csbsqlserver
+
+go 1.20
+
+require (
+	github.com/denisenkom/go-mssqldb v0.12.3
```
Comment (fnaranjo-vmw) [inline providers/terraform-provider-csbsqlserver/go.mod:6]: Do we want to favour Microsoft's fork of this module?
https://github.com/microsoft/go-mssqldb

It seems to be currently more active and I would expect it to have more strict collaboration guidelines and better security audits (although that's not necessarily true).

**Comment by fnaranjo-vmw** [comment]
Created a PR to replace the modules mentioned in my comments:
https://github.com/cloudfoundry/csb-brokerpak-aws/pull/1106

**Pull request merged**